### PR TITLE
Refactor DisplayPackageInfo: AttributionView

### PR DESCRIPTION
**Pull request by lennartclaas**
### Summary of changes

Apply the refactoring regarding DisplayPackageInfo to AttributionView as well as ReplaceAttributionPopup.

### Context and reason for change

Separate business and display logic by using DisplayPackageInfo in displaying components.

### CODE REVIEW

File: src/Frontend/Components/AttributionList/__tests__/AttributionList.test.tsx [hunk 1]
```diff
@@ -218,4 +232,82 @@ describe('The AttributionList', () => {
       true
     );
   });
+
+  // it('shows correct replace attribution buttons in the context menu', () => {
```
Comment (benedikt-richter) [inline src/Frontend/Components/AttributionList/__tests__/AttributionList.test.tsx:236]: why is this one commented out?

File: src/Frontend/Components/ReplaceAttributionPopup/ReplaceAttributionPopup.tsx [hunk 2]
```diff
@@ -59,11 +62,22 @@ export function ReplaceAttributionPopup(): ReactElement {
     dispatch(closePopup());
   }
 
+  const displayAttributions: Array<DisplayAttributionWithCount> =
+    Object.entries(attributions).map(([attributionId, packageInfo]) => {
```
Comment (benedikt-richter) [inline src/Frontend/Components/ReplaceAttributionPopup/ReplaceAttributionPopup.tsx:66]: Why was this popup modified?
Before, we just converted the attributions that we actually displayed, while now all attributions are converted and then we still only show two of these. Seems to be unnecessary compute.

File: src/Frontend/util/get-attribution-from-display-attribution-with-count.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  if (
+    attributionIdOrDisplayAttributionWithCount &&
+    isDisplayAttributionWithCount(attributionIdOrDisplayAttributionWithCount)
+  ) {
+    return attributionIdOrDisplayAttributionWithCount.attribution;
+  } else {
+    return {
+      attributionIds: [],
+    } as DisplayPackageInfo;
```
Comment (benedikt-richter) [inline src/Frontend/util/get-attribution-from-display-attribution-with-count.ts:31]: We should avoid type casts.

File: src/Frontend/util/get-attribution-from-display-attribution-with-count.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { DisplayPackageInfo } from '../../shared/shared-types';
+import {
+  DisplayAttributionWithCount,
+  isDisplayAttributionWithCount,
+} from '../types/types';
+
+export function getAttributionFromDisplayAttributionWithCount(
```
Comment (benedikt-richter) [inline src/Frontend/util/get-attribution-from-display-attribution-with-count.ts:12]: This is a duplicate of `getAttributionFromDisplayAttributionsWithCount`. Probably a relict of rebasing.

File: src/Frontend/Components/AttributionList/__tests__/AttributionList.test.tsx (hunk 1 above)
Comment (lennartclaas) [reply src/Frontend/Components/AttributionList/__tests__/AttributionList.test.tsx:236]: I just commented it out for reference and forgot to delete it. The apropriate test is above.

File: src/Frontend/Components/ReplaceAttributionPopup/ReplaceAttributionPopup.tsx (hunk 2 above)
Comment (lennartclaas) [reply src/Frontend/Components/ReplaceAttributionPopup/ReplaceAttributionPopup.tsx:66]: Good point, I reverted the changes.

File: src/Frontend/util/get-attribution-from-display-attribution-with-count.ts (hunk 3 above)
Comment (lennartclaas) [reply src/Frontend/util/get-attribution-from-display-attribution-with-count.ts:31]: Removed it as it does not seem to be needed anymore.

File: src/Frontend/util/get-attribution-from-display-attribution-with-count.ts (hunk 4 above)
Comment (lennartclaas) [reply src/Frontend/util/get-attribution-from-display-attribution-with-count.ts:12]: That's correct. I removed it.

**Review by benedikt-richter — APPROVED** [review]

**Pull request merged**